### DISCUSSION TITLE
Cannot Add Custom Plugin To Kong (v2) Container

### DISCUSSION THREAD

**Discussion by softwareeverything** [Kong/kong#8028] [Help]
Hi, I have two files under plugins folder on my local pc;

\
+++++++ Current dir on my local pc +++++++
plugins
----> handler.lua
----> schema.lua
\
\
Open cmd on Windows 10 then go to current dir;

> `docker exec -it --user root kong_container_id mkdir /usr/local/share/lua/5.1/kong/plugins/new_plugin`

>`docker cp plugins/. kong_container_id:/usr/local/share/lua/5.1/kong/plugins/new_plugin`

>`docker exec --user kong -e KONG_PLUGINS="bundled,new_plugin" -e KONG_LOG_LEVEL=debug kong_container_id kong reload -vv`

\
After all, I cannot see new_plugin on the results of http://localhost:8001

\
![image](https://user-images.githubusercontent.com/34370881/140031341-6ed1d3a1-2662-4525-9327-ed65037258cd.png)

\
\
And it gives an error message when I'm trying to add new_plugin to my service;

\
![image](https://user-images.githubusercontent.com/34370881/140031882-5b4e2dd9-420e-4f25-97f8-b07b166a57c0.png)

\
How to add my custom lua plugin to Kong v2 container? Thanks...

**Comment by tingeltangelthomas** [comment]
Hi @softwareeverything 
may https://docs.konghq.com/gateway-oss/2.6.x/plugin-development/distribution/#installing-the-plugin  will help you.
